sbc_compassion/models/correspondence_template.py

```diff
@@ -168,11 +172,72 @@ def _set_image(self):
                 })
                 self.template_attachment_id = attachment.id
 
-    @api.depends('template_image')
+    @api.onchange('template_image')
+    def _compute_size(self):
+        for template in self:
+            if template.template_image:
+                with tempfile.NamedTemporaryFile() as template_file:
+                    template_file.write(base64.b64decode(
+                        template.with_context(bin_size=False).template_image))
+                    template_file.flush()
+                    # Find the pattern inside the template image
+                    img = cv2.imread(template_file.name)
+                    template.page_height, template.page_width = img.shape[:2]
+                    template.qrcode_x_min = template.page_width * float(
+                        template.env['ir.config_parameter'].get_param(
+                            'qrcode_x_min'))
+                    template.qrcode_x_max = template.page_width * float(
+                        template.env['ir.config_parameter'].get_param(
+                            'qrcode_x_max'))
+                    template.qrcode_y_min = template.page_height * \
+                        float(template.env['ir.config_parameter'].get_param(
+                            'qrcode_y_min'))
+                    template.qrcode_y_max = template.page_height * \
+                        float(template.env['ir.config_parameter'].get_param(
+                            'qrcode_y_max'))
+
+    @api.depends('template_attachment_id',
+                 'pattern_image', 'template_image')
+    def _compute_template_keypoints(self):
+        """ This method computes all keypoints that can be automatically
+        detected (bluesquare and pattern_center)
+        """
+        for template in self:
+            if (template.template_image and template.pattern_image and
+                    _verify_template(template)):
+                with tempfile.NamedTemporaryFile() as template_file:
+                    template_file.write(base64.b64decode(
+                        template.with_context(bin_size=False).template_image))
+                    template_file.flush()
+                    # Find the pattern inside the template image
+                    img = cv2.imread(template_file.name)
+                    template.page_height, template.page_width = img.shape[:2]
```

Comment:
I do not remember why, but I think I had some troubles with theses fields when it was with the method compute. I will try again later.